packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/base-types.ts

```diff
@@ -625,6 +625,196 @@ export interface TransformResources {
+
+  /**
+   * Called when the ContainerDefinition type configured on Sagemaker Task
+   */
+  public bind(task: ISageMakerTask): ContainerDefinitionConfig {
+    return {
+      parameters: {
+        ContainerHostname: this.options.containerHostName,
+        Image: this.options.image?.bind(task).imageUri,
+        Mode: this.options.mode,
+        ModelDataUrl: this.options.modelS3Location?.bind(task, { forReading: true }).uri,
+        ModelPackageName: this.options.modelPackageName,
+        Environment: this.options.environmentVariables?.value,
+      },
+    };
+  }
+}
+
+/**
+ * Configuration of the container used to host the model
+ *
+ * @see https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ContainerDefinition.html
+ * @experimental
+ */
+export interface IContainerDefinition {
+  /**
+   * Called when the ContainerDefinition is used by a SageMaker task.
+   */
+  bind(task: ISageMakerTask): ContainerDefinitionConfig;
+}
+
+/**
+ * Configuration options for the ContainerDefinition
+ */
+export interface ContainerDefinitionConfig {
+  /**
+   * Additional parameters to pass to the base task
+   *
+   * @default - No additional parameters passed
+   */
+  readonly parameters?: { [key: string]: any };
+}
+
+/**
+ * Specifies how many models the container hosts
+ *
+ * @experimental
+ */
+export enum Mode {
+  /**
+   * Container hosts a single model
+   */
+  SINGLE_MODEL = 'SingleModel',
+  /**
+   * Container hosts multiple models
+   *
+   * @see https://docs.aws.amazon.com/sagemaker/latest/dg/multi-model-endpoints.html
+   */
+  MULTI_MODEL = 'MultiModel',
+}
+
+/**
+ * Identifies a model that you want to host and the resources to deploy for hosting it.
+ *
+ * @see  https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ProductionVariant.html
+ * @experimental
+ */
+export interface ProductionVariant {
+  /**
+   * The size of the Elastic Inference (EI) instance to use for the production variant.
+   *
+   * @default - None
+   */
+  readonly acceleratorType?: AcceleratorType;
+  /**
+   * Number of instances to launch initially.
+   */
+  readonly initialInstanceCount: number;
+  /**
+   * Determines initial traffic distribution among all of the models that you specify in the endpoint configuration.
+   *
+   * @default - 1.0
+   */
+  readonly initialVariantWeight?: number;
+  /**
+   * The ML compute instance type
+   */
+  readonly instanceType: ec2.InstanceType;
+  /**
+   * The name of the production variant.
+   */
+  readonly variantName: string;
+  /**
+   * The name of the model that you want to host. This is the name that you specified when creating the model.
+   */
+  readonly modelName: string;
+}
+
+/**
+ * The size of the Elastic Inference (EI) instance to use for the production variant.
+ * EI instances provide on-demand GPU computing for inference
+ *
+ * @see https://docs.aws.amazon.com/sagemaker/latest/dg/ei.html
+ * @experimental
+ */
+export enum AcceleratorType {
```

Comment:
it's probably okay for this PR, I'm wondering if we should have an enum like class with static constants, and an `of` method - i.e. here's an [example](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L570) in our `eks` module 

That way, if new accelerator types are supported, users won't have to wait for the CDK to update this enum and wait for subsequent releases.

Reply:
Changed AcceleratorType